tools/matRad_gammaIndex.m

```diff
@@ -0,0 +1,273 @@
+
+% init cube
+% cut all the zeros surrounding the interesting part
+cut1 = any(any(cube1,2),3);
+cut2 = any(any(cube1,1),3);
+cut3 = any(any(cube1,1),2);
+
+% avoids that "zero-slides" are deleted between two interest regions
+k = find(cut1); cut1(k(1):k(end)) = 1;
+k = find(cut2); cut2(k(1):k(end)) = 1;
+k = find(cut2); cut2(k(1):k(end)) = 1;
+
+% copy cubes
+cubex1c = cube1;
+cubex2c = cube2;
+
+% cut cubes
+cubex1c( ~cut1, :, :) = []; % rows
+cubex1c( :, ~cut2, :) = []; % columns
+cubex1c( :, :, ~cut3) = []; % slices  
+cubex2c( ~cut1, :, :) = []; % rows
+cubex2c( :, ~cut2, :) = []; % columns
+cubex2c( :, :, ~cut3) = []; % slices
+
+% pad cubes
+cubex1 = zeros([size(cubex1c,1)+2*searchX size(cubex1c,2)+2*searchY size(cubex1c,3)+2*searchZ]);
+cubex2 = zeros([size(cubex2c,1)+2*searchX size(cubex2c,2)+2*searchY size(cubex2c,3)+2*searchZ]);
+cubex1((1+searchX):(end-searchX), ...
+       (1+searchY):(end-searchY), ...
+       (1+searchZ):(end-searchZ)) = cubex1c;
+cubex2((1+searchX):(end-searchX), ...
+       (1+searchY):(end-searchY), ...
+       (1+searchZ):(end-searchZ)) = cubex2c;
+
+% interpolate if necessary
+if n > 0
+    cubex2 = interp3(cubex2,n,'cubic');
+end
+
+% set up temporary cubes required for calculation
+tmpCube = zeros(size(cubex1)); 
+tmpCube((1+searchX):(end-searchX), ...
+        (1+searchY):(end-searchY), ...
+        (1+searchZ):(end-searchZ)) = inf;
+
+
+gammaCubeSq = zeros(size(cubex1));
+gammaCubeSq((1+searchX):(end-searchX), ...
+            (1+searchY):(end-searchY), ...
+            (1+searchZ):(end-searchZ)) = inf;
+
+% find relevant indices for computation...
+ix = cubex1 > 0;
+
+% adjust dose threshold
+if strcmp(localglobal,'local')
+    doseThreshold = cubex1 .* relDoseThreshold/100;                
+elseif strcmp(localglobal,'global')
+    doseThreshold = relDoseThreshold/100 * max(cube1(:));
+end
+
+% search for min
+for i = -searchX:searchX
+    for j = -searchY:searchY
+        for k = -searchZ:searchZ
+            
+            delta_sq = ((i*resolution(1))^2 + ...
+                        (j*resolution(2))^2 + ...
+                        (k*resolution(3))^2) / dist2AgreeMm^2;                 
+            
+            tmpCube((1+searchX):(end-searchX), ...
+                    (1+searchY):(end-searchY), ...
+                    (1+searchZ):(end-searchZ)) = ...
+                             cubex1((1+searchX):(end-searchX), ...
+                                    (1+searchY):(end-searchY), ...
+                                    (1+searchZ):(end-searchZ)) ...
+                           - cubex2((1+((2^n)*searchX)+i) : 2^n : (end-((2^n)*searchX)+i), ...
+                                    (1+((2^n)*searchY)+j) : 2^n : (end-((2^n)*searchY)+j), ...
+                                    (1+((2^n)*searchZ)+k) : 2^n : (end-((2^n)*searchZ)+k));
+                    
+            tmpCube = tmpCube.^2 ./ doseThreshold.^2 + delta_sq;
+            
+            gammaCubeSq(ix) = min(gammaCubeSq(ix),tmpCube(ix));
```

Comment:
@pezzano Here comes a funny idea: I think we can compute the gamma index for different criteria simultaneously at reasonable computational overhead. Therefore we simply need to transform the criteria vector to an array and maybe turn the gammaCube into a cell array. Could you try this out?